src/platform/posix/os.c

```diff
@@ -45,8 +47,17 @@ int os_getopt(int argc, char * const argv[], const char *optstring)
 int32_t
 os_ncpus(void)
 {
-#ifdef _SC_NPROCESSORS_ONLN
+#if defined(_SC_NPROCESSORS_ONLN)
 	return sysconf(_SC_NPROCESSORS_ONLN);
+#elif defined(__APPLE__)
```

Comment:
also for BSD* : `defined (__FreeBSD__) || defined (__OpenBSD__) || \
   defined (__NetBSD__) || defined (__DragonFly__)`

in the EFL, we use that code:

```c

   int mib[4];
   int cpus;
   size_t len = sizeof(cpus);

   mib[0] = CTL_HW;
#ifdef HW_AVAILCPU
   mib[1] = HW_AVAILCPU;
#else
   mib[1] = HW_NCPU;
#endif
   sysctl(mib, 2, &cpus, &len, NULL, 0);
   if (cpus < 1)
      cpus = 1;
```